### PR TITLE
Adapt types for DSL

### PR DESCRIPTION
**Pull request by phanpak**

### CODE REVIEW

**Review by mikec — APPROVED** [review]

File: src/Types.ts [hunk 1]
```diff
@@ -24,11 +24,15 @@ export enum TokenStandard {
 
 export type BlockState = 'MINED' | 'NOT_MINED'
 
-export type NonceState = 'USED' | 'NOT_USED'
+export const USED_NONCE_STATE = 'USED';
+export const NOT_USED_NONCE_STATE = 'NOT_USED';
+export type NonceState = typeof USED_NONCE_STATE | typeof NOT_USED_NONCE_STATE;
```
Comment (mikec) [inline src/Types.ts:30]: @phanpak this seems a lot more complicated than just using strings. any reason why we need to do this?

Comment (mikec) [reply src/Types.ts:31]: @phanpak would this be equivalent?
```
export enum NonceState {
    USED,
    NOT_USED
}
```

**Review by mikec — CHANGES_REQUESTED** [review]

File: src/Types.ts [hunk 2]
```diff
@@ -165,11 +171,17 @@ export interface ConditionArgsBase {
   type: ConditionType
 }
 
+export interface TokenArgsForPriceCondition extends TokenArgs {
```
Comment (phanpak) [inline src/Types.ts:174]: it's for lack of a better name. 

I didn't know if we want to include decimals to TokenArgs directly? I thought it was very specific to that priceCondition DSL.

Comment (mikec) [reply src/Types.ts:174]: @phanpak let's add to TokenArgs directly

**Review by mikec — CHANGES_REQUESTED** [review]

File: src/Types.ts [hunk 3]
```diff
@@ -165,11 +171,17 @@ export interface ConditionArgsBase {
   type: ConditionType
 }
 
+export interface TokenWithDecimalsArgs extends TokenArgs {
+  decimals: number;
```
Comment (mikec) [inline src/Types.ts:175]: @phanpak decimals is not a number. number is floating point like 12.34. decimals should be `BigIntish`

**Review by mikec — CHANGES_REQUESTED** [review]

**Review by mikec — APPROVED** [review]

**Pull request merged**